Advanced-Recruit-Training/src/LogFileReader.cpp

```diff
@@ -47,25 +48,28 @@ bool LogFileReader::readAll(const QString& fileName)
             emit batteryDataReceived(batteryData);
         }
     }
-
     return true;
 }
 
-/* File input is a csv file in the format of hh:mm:ss:zzz, voltage, current.
- * Negative current values denote a charging battery.
- * Need to implement error checking for the correct number of values and
- * that the conversion from string to double is sucessful.*/
 bool LogFileReader::parseLine(const QString& line, BatteryData& batteryData) const
 {
     QStringList sections = line.split(BATDATA_DELIMITER);
-
+    if(sections.size() != 3 || sections.at(0).isNull() || sections.at(1).isNull() || sections.at(2).isNull())
```

Comment:
This is actually a bit redundant. If an element is null, the size of sections wouldn't be 3!